unlock-app/src/components/interface/connect/ConnectingWaas.tsx

```diff
@@ -0,0 +1,113 @@
+import { Placeholder } from '@unlock-protocol/ui'
+import { signOut, useSession } from 'next-auth/react'
+import { useRouter } from 'next/router'
+import { useEffect, useRef } from 'react'
+import { config } from '~/config/app'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useAuthenticate } from '~/hooks/useAuthenticate'
+import { useSIWE } from '~/hooks/useSIWE'
+import WaasProvider from '~/services/WaasProvider'
+
+export type ConnectingWaasProps = {
+  shouldReloadOnTimeout?: boolean
+}
+
+export const ConnectingWaas = ({
+  shouldReloadOnTimeout = false,
+}: ConnectingWaasProps) => {
+  const { data: session } = useSession()
+  const { authenticateWithProvider } = useAuthenticate()
+  const { signIn: siweSignIn, isSignedIn } = useSIWE()
+
+  const { connected } = useAuth()
+
+  const router = useRouter()
+  const restoredState = JSON.parse(
+    decodeURIComponent((router.query.state as string) || '{}')
+  )
+
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null)
+
+  const redirect = () => {
+    if (restoredState.redirectUrl) {
+      router.push(restoredState.redirectUrl)
+    }
+  }
+
+  useEffect(() => {
+    if (!timeoutRef.current) {
+      timeoutRef.current = setTimeout(async () => {
```

Comment:
What's the purpose of the timeout here? I assume it should _never_ happen, right?

Reply:
Is that happening?